desktop/components/react/utils/template_renderer.js

```diff
@@ -0,0 +1,41 @@
+import invariant from 'invariant'
+import jade from 'jade'
+import path from 'path'
+
+import {
+  isArray,
+  isString
+} from 'underscore'
+
+export default function templateRenderer (templates, options = {}) {
+  const isValid = isArray(templates) || isString(templates)
+
+  invariant(isValid,
+    '(lib/template_renderer.js) ' +
+    'Error rendering templates: `templates` must be a string or array of ' +
+    'strings representing the path to the template'
+  )
+
+  templates = isArray(templates)
+    ? templates
+    : [templates]
+
+  const basePath = options.basePath || ''
+
+  const templateFns = templates.map(file => {
+    return jade.compileFile(path.join(basePath, file), {
+      cache: true,
+      ...options
+    })
+  })
+
+  return {
```

Comment:
At least for now, it looks like we only ever immediately call `render` after `templateRenderer`. I was guessing that the `cache: true` probably prevents this function from being expensive after the first call. I think this is a place where I'd error on the side of [YAGNI](https://en.wikipedia.org/wiki/You_aren't_gonna_need_it), and then just make a partially lazy version later, if a use case pops up.